refreshable/src/main/scala/prometheus4cats/refreshable/InstrumentedRefreshable.scala

```diff
@@ -276,4 +385,40 @@ object InstrumentedRefreshable {
           }
     }
 
+  def fromExisting[F[_]: MonadCancelThrow, A](
+      refreshable: Refreshable[F, A],
+      name: String,
+      metricFactory: MetricFactory.WithCallbacks[F]
+  ): Resource[F, InstrumentedRefreshable[F, A]] =
+    instanceMetrics[F, A](metricFactory).toResource
+      .flatTap(_ => callback(name, refreshable, metricFactory))
+      .evalMap { case (readCounter, runningGauge, exhaustedRetriesGauge) =>
+        runningGauge
+          .set(true, name)
+          .as {
+            // Set the running gauge on read as we can't do this via callback or from the stream
+            val setRunningGauge = MonadCancelThrow[F].uncancelable { poll =>
+              poll(
```

Comment:
What is this `uncancelable` doing? It looks like the whole interior is unmasked again via `poll`?

Reply:
Ooops I made everything cancellable 🤦